source/beamline/BioXAS/BioXASSideTableCoordinator.cpp

```diff
@@ -0,0 +1,414 @@
+
+	if (!verticalUpstreamControl_->withinTolerance(targetUpstreamHeight)) {
+		verticalUpstreamControl_->move(targetUpstreamHeight);
+	}
+	if (!verticalDownstreamControl_->withinTolerance(targetDownstreamHeight)) {
+		verticalDownstreamControl_->move(targetDownstreamHeight);
+	}
+}
+
+void BioXASSideTableCoordinator::onPitchStopControlValueChanged(double value)
+{
+	Q_UNUSED(value)
+	if(!connectedOnce_)
+		return;
+
+	if (softIOCPitchStopControl_->withinTolerance(1)) {
+		verticalUpstreamControl_->stop();
+		verticalDownstreamControl_->stop();
+
+		softIOCPitchStopControl_->move(0); // restore the value to 0
+	}
+}
+
+void BioXASSideTableCoordinator::onLateralControlValueChanged(double value)
+{
+	Q_UNUSED(value)
+	if(!connectedOnce_)
+		return;
+
+	double delta = value - softIOCLateralFeedbackControl_->value();
+
+	double horizontalUpstreamPosition = horizontalUpstreamControl_->value() - delta;
+	double horizontalDownstreamPosition = horizontalDownstreamControl_->value() + delta;
+
+	if (!horizontalUpstreamControl_->withinTolerance(horizontalUpstreamPosition)) {
+		horizontalUpstreamControl_->move(horizontalUpstreamPosition);
+	}
+	if (!horizontalDownstreamControl_->withinTolerance(horizontalDownstreamPosition)) {
+		horizontalDownstreamControl_->move(horizontalDownstreamPosition);
+	}
+}
+
+void BioXASSideTableCoordinator::onLateralStopControlValueChanged(double value)
+{
+	Q_UNUSED(value)
+	if(!connectedOnce_)
+		return;
+
+	if (softIOCLateralStopControl_->withinTolerance(1)) {
+		horizontalUpstreamControl_->stop();
+		horizontalDownstreamControl_->stop();
+
+		softIOCLateralStopControl_->move(0); // restore the value to 0
+	}
+}
+
+void BioXASSideTableCoordinator::onYawControlValueChanged(double angle)
+{
+	Q_UNUSED(angle)
+	if(!connectedOnce_)
+		return;
+
+	double delta = tableHorizontalMotorPosition_ * tan(degreeToRadian(angle));
+
+	if (!horizontalUpstreamControl_->withinTolerance(-delta)) {
+		horizontalUpstreamControl_->move(-delta);
+	}
+	if (!horizontalDownstreamControl_->withinTolerance(delta)) {
+		horizontalDownstreamControl_->move(delta);
+	}
+}
+
+void BioXASSideTableCoordinator::onYawStopControlValueChanged(double value)
+{
+	Q_UNUSED(value)
+	if(!connectedOnce_)
+		return;
+
+	if (softIOCYawStopControl_->withinTolerance(1)) {
+		horizontalUpstreamControl_->stop();
+		horizontalDownstreamControl_->stop();
+
+		softIOCYawStopControl_->move(0); // restore the value to 0
+	}
+}
+
+void BioXASSideTableCoordinator::manipulateVerticalPVChange(double upstreamHeight, double downstreamHeight, AMControl* heightPV, AMControl *pitchPV)
+{
+	// The vertical height is the height of the center of the table, which is the average of upstream value and the downstream value
+	double height = (upstreamHeight + downstreamHeight) / 2;
+	// The pitch is the degree of the arcTangent value of (dH - uH) / distance
+	double pitch =  radianToDegree(atan((downstreamHeight - upstreamHeight) / (tableVerticalMotorPosition_ * 2)));
```

Comment:
Are these all relative movements?  Because I think you are going to have issues with these within tolerance checks.